lib/core/components/config.js

```diff
@@ -58,7 +58,12 @@ var _class = function () {
     this.logVerbosity = setup.logVerbosity || false;
     this.suppressLeaveEvents = setup.suppressLeaveEvents || false;
 
-    this.announceFailedHeartbeats = setup.announceFailedHeartbeats || true;
+    if (setup.announceFailedHeartbeats != null) {
```

Comment:
```suggestion

    if (setup.announceFailedHeartbeats  !== null) {
```

Reply:
Sorry, I didn't understand the full code but it seems its better to use !== in if condition than != 
(undefined != null returns false while undefined !== null returns true)